polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp

```diff
@@ -1207,9 +1207,28 @@ MemoryAccessAnalysis::getMemoryAccess(const MemRefAccess &access) const {
   return it->second;
 }
 
+unsigned MemoryAccessAnalysis::getGridSize(FunctionOpInterface func) const {
+  if (func.getNumArguments() == 0)
+    return 0;
+
+  Value lastArg = func.getArguments().back();
+  if (!isa<MemRefType>(lastArg.getType()))
+    return 0;
+
+  Type elemTy = cast<MemRefType>(lastArg.getType()).getElementType();
+  if (auto ndItemTy = dyn_cast<sycl::NdItemType>(elemTy))
+    return ndItemTy.getDimension();
+  if (auto itemTy = dyn_cast<sycl::ItemType>(elemTy))
+    return itemTy.getDimension();
```

Comment:
use TypeSwitch ?

Reply:
We should also assert the value is in 1, 2 or 3 for correctness.

Reply:
Is zero not allowed ?

Reply:
ok checked zero isn't allowed.

Reply:
zero is not allowed, it is allowed for accessor.